lib/inspec/secrets/yaml.rb

```diff
@@ -9,15 +9,19 @@ class YAML < Inspec.secrets(1)
     attr_reader :attributes
 
     def self.resolve(target)
-      unless target.is_a?(String) && File.file?(target) && target.end_with?('.yml')
+      unless target.is_a?(String) && File.file?(target)
         return nil
       end
       new(target)
     end
 
     # array of yaml file paths
     def initialize(target)
-      @attributes = ::YAML.load_file(target)
+      begin
```

Comment:
Travis is currently red because of a lint issue here. You've done excellent here catching a potential exception of loading in invalid YAML, however, you don't need to do a `begin` when you're in a method definition if you're essentially catching the entire method body. This can be shortened to:

```
def initialize(target)
  @attributes = ::YAML.load_file(target)
rescue => e
  raise "Error reading Inspec attributes: #{e}"
end
```

That should clear up the failed travis tests.